File: cadastro-moradores-api/src/main/java/com/elbertribeiro/cadastromoradores/usuario/Usuario.java
package com.elbertribeiro.cadastromoradores.usuario;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String nome;
    private String email;
    private String senha;
    private LocalDate dataCriacao = LocalDate.now();
    private String token;
    private String telefone;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

    public LocalDate getDataCriacao() {
        return dataCriacao;
    }

    public void setDataCriacao(LocalDate dataCriacao) {
        this.dataCriacao = dataCriacao;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }
}

File: cadastro-moradores-api/src/main/kotlin/com/elbertribeiro/cadastromoradores/usuario/Usuario.kt
package com.elbertribeiro.cadastromoradores.usuario

import com.elbertribeiro.cadastromoradores.telefone.Telefone
import com.elbertribeiro.cadastromoradores.telefone.toTelefoneDto
import java.time.LocalDate
import java.util.stream.Collectors
import javax.persistence.*

@Entity
class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    val id: Long = 0
    val nome: String = ""
    val email: String = ""
    val senha: String = ""
    val dataCriacao: LocalDate = LocalDate.now()
    val token: String = ""

    @OneToMany(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE])
    val telefones: List<Telefone>? = null
}

fun Usuario.toUsuarioDto() = UsuarioDto(
    nome, email, senha, token, telefones = telefones?.stream()?.map { telefone -> telefone.toTelefoneDto() }?.collect(
        Collectors.toList())
)